src/newgrf_object.cpp

```diff
@@ -305,7 +305,7 @@ static uint32_t GetCountAndDistanceOfClosestInstance(byte local_id, uint32_t grf
 	switch (variable) {
 		/* Relative position. */
 		case 0x40: {
-			uint offset = this->tile - this->obj->location.tile;
+			uint offset = static_cast<uint32_t>(this->tile - this->obj->location.tile);
```

Comment:
Could this work instead? TileX() and TileY() already take a TileIndex as an argument.

```suggestion
			TileIndex offset = this->tile - this->obj->location.tile;
```